fds/services/dvc_service.py

```diff
@@ -228,3 +229,71 @@ def push(remote: str) -> Any:
+            remote_name_with_url = config_list.split("\t")
+            if (len(remote_name_with_url) == 2):
+                config_list_dict[remote_name_with_url[0]] = str(remote_name_with_url[1])
+        return config_list_dict
+
+    @staticmethod
+    def _show_choice_of_remotes(remotes: dict) -> str:
+        choices = list(remotes.keys())
+        choices.append("Cancel Pull")
+        questions = [
+            {
+                'type': 'list',
+                'name': 'remote',
+                'message': 'Choose the remote to use for pulling dvc repo?',
+                'choices': choices
+            }
+        ]
+        answers = PyInquirer.prompt(questions)
+        return answers["remote"]
+
+    def pull(self, git_url: str, remote_url_or_name: Optional[str]) -> Any:
+        """
+        Responsible for pulling the latest changes from DVC remote based on dvc.yaml and .dvc files
+        :param git_url: The git url provided
+        :param remote_url_or_name: Optional Remote dvc url or name to pull the dvc repository
+        :return:
+        """
+        if remote_url_or_name is None:
+            #If nothing is specified
+            #First check if its dagshub repo
+            if "dagshub.com" in git_url.lower():
+                # then construct a dagshub url from the git url
+                dvc_url = construct_dvc_url_from_git_url_dagshub(git_url)
+                # find it from the remote
+                remote_list = DVCService.__get_remotes_list()
+                for remote, url in remote_list.items():
+                    if url == dvc_url:
+                        remote_url_or_name = remote
+                if remote_url_or_name is None:
+                    # if url is not in remote, then add it to remote and use that remote
+                    remote_url_or_name="dagshub"
+                    execute_command(["dvc", "remote", "add", remote_url_or_name, dvc_url])
+            else:
+                # If its not dagshub url, then check if there exists a default remote
+                default_remote_cmd = execute_command(["dvc", "remote", "default"], capture_output=True)
+                default_remote = convert_bytes_to_string(default_remote_cmd.stdout).strip()
+                if default_remote == "" or "No default remote set":
+                    # No default remote defined
+                    # So show all the remotes to user and let user choose
+                    remote_list = DVCService.__get_remotes_list()
+                    remote_url_or_name=DVCService._show_choice_of_remotes(remote_list)
+                    # If the user chooses to cancel pull
+                    if remote_url_or_name not in remote_list:
+                        return 0
+                else:
+                    remote_url_or_name=default_remote
+
+        if is_url(remote_url_or_name):
+            execute_command(["dvc", "get", remote_url_or_name, get_dvc_repo_name_from_url(remote_url_or_name)], capture_output=False)
```

Comment:
This isn't actually how `dvc get` works
https://dvc.org/doc/command-reference/get

I think for this to work, the only way would be to run something like:
```python
remote_uuid = ....
dvc remote add --local remote_uuid remote_url_or_name
dvc pull -r remote_uuid
dvc remote remove --local remote_uuid
```

Reply:
Maybe the most reasonable thing for now is to not support a URL in the argument

Reply:
okay, I'll remove the support for url